src/react/reconcilation.js

```diff
@@ -1174,6 +1184,21 @@ export class Reconciler {
       );
       return reactElement;
     }
+
+    // If we have a new result and we might have a key value then wrap our inlined result in a
+    // `<React.Fragment key={keyValue}>` so that we may maintain the key.
+    if (needsKey && keyValue.mightNotBeNull()) {
+      const react = this.realm.fbLibraries.react;
```

Comment:
Can we extract this key adding logic into its own function that has a descriptive name. We might want to apply the same logic in other parts of the reconciler or in `branching.js` at some point so it would be useful to have as utility function.